src/app/datepicker/datepicker.directive.view.spec.ts

```diff
@@ -0,0 +1,473 @@
+import { async, fakeAsync, TestBed, tick } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+
+import { buildComponent, MzTestWrapperComponent } from '../shared/test-wrapper';
+import { MzDatepickerContainerComponent, MzDatepickerDirective } from './';
+
+describe('MzDatepickerDirective:view', () => {
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        FormsModule,
+      ],
+      declarations: [
+        MzDatepickerContainerComponent,
+        MzDatepickerDirective,
+        MzTestWrapperComponent,
+      ],
+    });
+  }));
+
+  describe('label', () => {
+    let nativeElement: any;
+
+    function label(): HTMLLabelElement {
+      return nativeElement.querySelector('label');
+    }
+
+    it('should be shown correctly when provided', async(() => {
+
+      buildComponent<any>(`
+        <mz-datepicker-container>
+          <input mz-datepicker
+            id="datepicker"
+            [label]="'label-x'">
+        </mz-datepicker-container>
+      `).then((fixture) => {
+        nativeElement = fixture.nativeElement;
+        fixture.detectChanges();
+
+        expect(label().innerHTML).toBe('label-x');
+      });
+    }));
+
+    it('should be empty when omitted', async(() => {
+
+      buildComponent<any>(`
```

Comment:
You can forget my previous comment about datepicker without label. This test is testing this use case.

Reply:
Alright ... you know you can delete comments right? 😉